ding/model/template/efficientzero/efficientzero_tictactoe_model.py

```diff
@@ -0,0 +1,607 @@
+        self.delta = delta
+
+
+value_support = DiscreteSupport(-10, 10, delta=1)
+reward_support = DiscreteSupport(-10, 10, delta=1)
+
+
+def inverse_reward_transform(reward_logits):
+    return inverse_scalar_transform(reward_logits, reward_support)
+    # return reward_logits  # TODO
+
+
+def inverse_value_transform(value_logits):
+    return inverse_scalar_transform(value_logits, value_support)
+    # return value_logits  # TODO
+
+
+def inverse_scalar_transform(logits, scalar_support):
+    """ Reference from MuZerp: Appendix F => Network Architecture
+    & Appendix A : Proposition A.2 in https://arxiv.org/pdf/1805.11593.pdf (Page-11)
+    """
+    value_support = DiscreteSupport(-10, 10, delta=1)
+    reward_support = DiscreteSupport(-10, 10, delta=1)
+    delta = value_support.delta
+    value_probs = torch.softmax(logits, dim=1)
+    value_support = torch.ones(value_probs.shape)
+    value_support[:, :] = torch.from_numpy(np.array([x for x in scalar_support.range]))
+    value_support = value_support.to(device=value_probs.device)
+    value = (value_support * value_probs).sum(1, keepdim=True) / delta
+
+    epsilon = 0.001
+    sign = torch.ones(value.shape).float().to(value.device)
+    sign[value < 0] = -1.0
+    output = (((torch.sqrt(1 + 4 * epsilon * (torch.abs(value) + 1 + epsilon)) - 1) / (2 * epsilon)) ** 2 - 1)
+    output = sign * output * delta
+
+    nan_part = torch.isnan(output)
+    output[nan_part] = 0.
+    output[torch.abs(output) < epsilon] = 0.
+    return output
+
+
+# value=torch.randn([2,32])
+# inverse_value_transform(value)
+
+
+def mlp(
+    input_size,
+    layer_sizes,
+    output_size,
+    output_activation=nn.Identity,
+    activation=nn.ReLU,
+    momentum=0.1,
+    init_zero=False,
+):
+    """MLP layers
+    Parameters
+    ----------
+    input_size: int
+        dim of inputs
+    layer_sizes: list
+        dim of hidden layers
+    output_size: int
+        dim of outputs
+    init_zero: bool
+        zero initialization for the last layer (including w and b).
+        This can provide stable zero outputs in the beginning.
+    """
+    print(input_size, layer_sizes, output_size)
+    sizes = [input_size] + layer_sizes + [output_size]
+    layers = []
+    for i in range(len(sizes) - 1):
+        if i < len(sizes) - 2:
+            act = activation
+            layers += [nn.Linear(sizes[i], sizes[i + 1]), nn.BatchNorm1d(sizes[i + 1], momentum=momentum), act()]
+        else:
+            act = output_activation
+            layers += [nn.Linear(sizes[i], sizes[i + 1]), act()]
+
+    if init_zero:
+        layers[-2].weight.data.fill_(0)
+        layers[-2].bias.data.fill_(0)
+
+    return nn.Sequential(*layers)
+
+
+# test mlp
+# fc = mlp(27, [64,32,16,16], 9)
+# print(fc)
+
+
+def conv3x3(in_channels, out_channels, stride=1):
+    return nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=stride, padding=1, bias=False)
+
+
+# Residual block
+class ResidualBlock(nn.Module):
```

Comment:
use our residual block implementation, if there are some mismatch parts, modify our codes adapted to the demand of muzero/efficientzero